src/branch.rs

```diff
@@ -28,6 +28,16 @@ impl<'repo> Branch<'repo> {
         Branch { inner: reference }
     }
 
+    /// Ensure the branch name is well-formed.
+    pub fn name_is_valid(name: &str) -> Result<bool, Error> {
+        let name = CString::new(name).unwrap();
```

Comment:
Unwrap will panic at runtime. You can check other such CString usages in here but you can essentially propagate the error up by using `?` instead

Reply:
ah, didn't know about `?`... that seems like a really nice thing to use! also didn't know that unwrap panics. thanks for the feedback!